snippets/isArmstrongNumber.md

```diff
@@ -0,0 +1,17 @@
+### isArmstrongNumber
+
+Checks if the given number is an armstrong number or not.
+
+Convert the given number into array of digits. Use `Math.pow()` to get the appropriate power for each digit and sum them up. If the sum is equal to the number itself, return `true` otherwise `false`.
+
+```js
+const isArmstrongNumber = digits => {
+	let total = 0, arr = (digits+"").split("");
+	arr.map(d => total += Math.pow(parseInt(d), arr.length))
```

Comment:
This could be a reduce instead, saving the deceleration of total, and making the comparison
```js
return arr.reduce( ( a, d ) => a + Math.pow( parseInt( d ), arr.length ) ) == digits ? true : false
```

Reply:
Of course then you could also use the internal closure trick to declare the arr too, but that gets pretty obtuse for most people. While fun, it often leads to a lot of parenthesis

Reply:
need to do `parseInt(a)` before adding...